internal/predicate/nonadminbackup_predicate.go

```diff
@@ -38,49 +39,67 @@ func getNonAdminBackupPredicateLogger(ctx context.Context, name, namespace strin
 
 // Create event filter
 func (NonAdminBackupPredicate) Create(ctx context.Context, evt event.CreateEvent) bool {
-	if _, ok := evt.Object.(*nacv1alpha1.NonAdminBackup); ok {
-		nameSpace := evt.Object.GetNamespace()
-		name := evt.Object.GetName()
-		logger := getNonAdminBackupPredicateLogger(ctx, name, nameSpace)
-		logger.V(1).Info("Received Create NonAdminBackupPredicate")
-		return true
+	nameSpace := evt.Object.GetNamespace()
+	name := evt.Object.GetName()
+	logger := getNonAdminBackupPredicateLogger(ctx, name, nameSpace)
+	logger.V(1).Info("NonAdminBackupPredicate: Received Create event")
+	if nonAdminBackup, ok := evt.Object.(*nacv1alpha1.NonAdminBackup); ok {
+		if nonAdminBackup.Status.Phase == constant.EmptyString || nonAdminBackup.Status.Phase == nacv1alpha1.NonAdminBackupPhaseNew {
+			logger.V(1).Info("NonAdminBackupPredicate: Accepted Create event")
+			return true
+		}
 	}
-
+	logger.V(1).Info("NonAdminBackupPredicate: Rejecting Create event")
 	return false
 }
 
 // Update event filter
 func (NonAdminBackupPredicate) Update(ctx context.Context, evt event.UpdateEvent) bool {
-	if _, ok := evt.ObjectNew.(*nacv1alpha1.NonAdminBackup); ok {
-		nameSpace := evt.ObjectNew.GetNamespace()
-		name := evt.ObjectNew.GetName()
-		logger := getNonAdminBackupPredicateLogger(ctx, name, nameSpace)
-		logger.V(1).Info("Received Update NonAdminBackupPredicate")
+	// Do not reconcile on Status update
+	nameSpace := evt.ObjectNew.GetNamespace()
+	name := evt.ObjectNew.GetName()
+	logger := getNonAdminBackupPredicateLogger(ctx, name, nameSpace)
+	logger.V(1).Info("NonAdminBackupPredicate: Received Update event")
+
+	if evt.ObjectNew.GetGeneration() != evt.ObjectOld.GetGeneration() {
+		logger.V(1).Info("NonAdminBackupPredicate: Accepted Update event - generation change")
 		return true
 	}
+
+	if nonAdminBackupOld, ok := evt.ObjectOld.(*nacv1alpha1.NonAdminBackup); ok {
+		if nonAdminBackupNew, ok := evt.ObjectNew.(*nacv1alpha1.NonAdminBackup); ok {
+			oldPhase := nonAdminBackupOld.Status.Phase
+			newPhase := nonAdminBackupNew.Status.Phase
+
+			// New phase set, reconcile
+			if oldPhase == constant.EmptyString && newPhase != constant.EmptyString {
```

Comment:
could not this if else be simplified to 
```go
if oldPhase != newPhase {
    	logger.V(1).Info("MESSAGE")
	return true
}
```

Reply:
No, because that would mean for any oldPhase that is not a newPhase but we want to have:

New Object (EMPTY oldPhase) ---> AcceptUpdate --> New
New ----> Created

With your logic we would also accept:

Created ---> New
or
Unknown ---> Created

This was the biggest challenge here, because when the controller starts we are getting not `Create` events on all objects, however the same objects when the controller is running may get `Update` event, so we need to be very careful on what situations do we want to reconcile, to not reconscile on the objects that are in state which we should consider "already in progress" by Velero.